tests/test_nodes.py

```diff
@@ -499,14 +500,31 @@ def test_deepcopy(obj: Any) -> None:
             True,
         ),
         (EnumNode(enum_type=Enum1, value=Enum1.BAR), Enum1.BAR, True),
+        (InterpolationResultNode("foo"), "foo", True),
+        (InterpolationResultNode("${foo}"), "${foo}", True),
+        (InterpolationResultNode("${foo"), "${foo", True),
+        (InterpolationResultNode(None), None, True),
+        (InterpolationResultNode(1), 1, True),
+        (InterpolationResultNode(1.0), 1.0, True),
+        (InterpolationResultNode(True), True, True),
+        (InterpolationResultNode(Color.RED), Color.RED, True),
+        (InterpolationResultNode({"a": 0, "b": 1}), {"a": 0, "b": 1}, True),
+        (InterpolationResultNode([0, None, True]), [0, None, True], True),
+        (InterpolationResultNode("foo"), 100, False),
+        (InterpolationResultNode(100), "foo", False),
     ],
 )
 def test_eq(node: ValueNode, value: Any, expected: Any) -> None:
     assert (node == value) == expected
     assert (node != value) != expected
     assert (value == node) == expected
     assert (value != node) != expected
-    assert (node.__hash__() == value.__hash__()) == expected
+
+    try:
+        assert (node.__hash__() == value.__hash__()) == expected
+    except TypeError as exc:
+        # Skip this check if unhashable.
+        assert "unhashable type" in str(exc)
```

Comment:
If the problem is that InterpolationResultNode is not hashable, be more specific:
```suggestion
    if not isinstance(node, InterpolationResultNode):
        assert (node.__hash__() == value.__hash__()) == expected
```

Alternatively you can fix InterpolationResultNode to return hash of the value.

Reply:
No, the problem is when wrapping a dict/list which aren't hashable. I changed it in c4004bf to make it less generic but more explicit.